File: RESTFul/AAB/Java/app1/src/main/java/com/att/api/rest/HttpPatch.java
/* vim: set expandtab tabstop=4 shiftwidth=4 softtabstop=4: */

/*
 * ====================================================================
 * LICENSE: Licensed by AT&T under the 'Software Development Kit Tools
 * Agreement.' 2014.
 * TERMS AND CONDITIONS FOR USE, REPRODUCTION, AND DISTRIBUTIONS:
 * http://developer.att.com/sdk_agreement/
 *
 * Copyright 2014 AT&T Intellectual Property. All rights reserved.
 * For more information contact developer.support@att.com
 * ====================================================================
 */

package com.att.api.rest;

import java.net.URI;
import org.apache.http.client.methods.HttpEntityEnclosingRequestBase;

// http://tools.ietf.org/html/rfc5789
public class HttpPatch extends HttpEntityEnclosingRequestBase {
    public HttpPatch(final String uri) {
        super();
        setURI(URI.create(uri));
    }

    @Override
    public String getMethod() {
        return "PATCH";
    }

}

File: RESTFul/SpeechCustom/PHP/app1/lib/Restful/HttpPatch.php
<?php
namespace Att\Api\Restful;

class HttpPatch
{

    /**
     * PATCH body, if applicable.
     */
    private $_body;

    public function __construct($body)
    {
        $this->_body = $body;
    }

    public function getBody()
    {
        return $this->_body;
    }
}

?>
